src/dep/cratesio.rs

```diff
@@ -0,0 +1,98 @@
+use std::path::{self, Component, Path as StdPath, PathBuf};
+
+use colored::Colorize as _;
+
+#[derive(Debug, PartialEq)]
+pub(crate) struct Path<'p> {
```

Comment:
Why are you using `pub(crate)`? Could you not just use `pub`, since it is a application anyways?

Reply:
mainly out of habit: if it's not going to be used from outside the crate then I use `pub(crate)`. Also, rust-analyzer has a code assist to add `pub(crate)`, but not to add `pub` :wink:

Reply:
> if it's not going to be used from outside the crate

in binary crates, the API is not meant to be used by other crates so I `pub(crate)` all the things

Reply:
> in binary crates, the API is not meant to be used by other crates so I `pub(crate)` all the things

but using a application API is not possible even if it would be `pub`, is it? 👀

Reply:
> Also, rust-analyzer has a code assist to add `pub(crate)`, but not to add `pub` wink

I can see this reason 😉